conans/client/conf/detect.py

```diff
@@ -146,9 +159,9 @@ def _get_default_compiler(output):
     if cc or cxx:  # Env defined, use them
         output.info("CC and CXX: %s, %s " % (cc or "None", cxx or "None"))
         command = cc or cxx
-        if "gcc" in command:
+        if _is_real_compiler("gcc", command):
```

Comment:
I think it is clearer to get here the output of the "command --help", the "if "gcc" in output:" and "if "clang" in output". The function is real compiler feels a little confusing. Maybe also check the output in lowercase in case GCC or Clang in output.

Reply:
Sorry for my comments formatting. I'm with the mobile.

Reply:
Yes, agree, I'd think it would be more understandable. I think that @lasote meant ``command --version``, not ``--help``.

Reply:
@lasote  do you mean something like this?

```python
if "gcc" in command:
    _, out = _execute("%s --version", command)
    if "gcc" in out.lower():
         return _gcc_compiler(output, command)
if "clang" in command.lower():
   _, out =  _execute("%s --version", command.lower())
   if "clang" in out.lower():
        return _clang_compiler(output, command)   
```

Reply:
I've fixed it with 2nd option